postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresErrorDetailsProvider.java

```diff
@@ -0,0 +1,93 @@
+import io.cdap.plugin.db.DBErrorDetailsProvider;
+import io.cdap.plugin.util.DBUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A custom ErrorDetailsProvider for Postgres plugins.
+ */
+public class PostgresErrorDetailsProvider extends DBErrorDetailsProvider {
+  // https://www.postgresql.org/docs/current/errcodes-appendix.html
+  private static final Map<String, ErrorType> ERROR_CODE_TO_ERROR_TYPE;
+  private static final Map<String, ErrorCategory> ERROR_CODE_TO_ERROR_CATEGORY;
+  static {
+    ERROR_CODE_TO_ERROR_TYPE = new HashMap<>();
+    ERROR_CODE_TO_ERROR_TYPE.put("01", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("02", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("08", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("0A", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("22", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("23", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("28", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("40", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("42", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("53", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("54", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("55", ErrorType.USER);
+    ERROR_CODE_TO_ERROR_TYPE.put("57", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("58", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("P0", ErrorType.SYSTEM);
+    ERROR_CODE_TO_ERROR_TYPE.put("XX", ErrorType.SYSTEM);
+
+    ErrorCategory.ErrorCategoryEnum plugin = ErrorCategory.ErrorCategoryEnum.PLUGIN;
+    ERROR_CODE_TO_ERROR_CATEGORY = new HashMap<>();
+    ERROR_CODE_TO_ERROR_CATEGORY.put("01", new ErrorCategory(plugin, "Warning"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("02", new ErrorCategory(plugin, "No Data"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("08", new ErrorCategory(plugin, "Connection Exception"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("0A", new ErrorCategory(plugin, "Feature Not Supported"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("22", new ErrorCategory(plugin, "Data Exception"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("23", new ErrorCategory(plugin, "Integrity Constraint Violation"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("28", new ErrorCategory(plugin, "Invalid Authorization Specification"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("40", new ErrorCategory(plugin, "Transaction Rollback"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("42", new ErrorCategory(plugin, "Syntax Error or Access Rule Violation"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("53", new ErrorCategory(plugin, "Insufficient Resources"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("54", new ErrorCategory(plugin, "Program Limit Exceeded"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("55", new ErrorCategory(plugin, "Object Not in Prerequisite State"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("57", new ErrorCategory(plugin, "Operator Intervention"));
+    ERROR_CODE_TO_ERROR_CATEGORY.put("58", new ErrorCategory(plugin, "System Error"));
```

Comment:
`Postgres Server System Error` to clearly indicate `System Error` from postgres server.

Reply:
similar comment for all.

Reply:
Updated !